pkg/sloop/queries/timerange.go

```diff
@@ -36,7 +36,24 @@ const minLookback = 1 * time.Minute
 // TODO: If wall clock is in the middle of the newest partition min-max time we can use it
 func computeTimeRange(params url.Values, tables typed.Tables, maxLookBack time.Duration) (time.Time, time.Time, error) {
 	endOfTime := getEndOfTime(tables)
-	return computeTimeRangeInternal(params, endOfTime, maxLookBack)
+	actualEndTime := computeActualEndTime(params.Get(EndTimeParam), endOfTime)
+	return computeTimeRangeInternal(params, actualEndTime, maxLookBack)
+}
+
+// Compare result of function `getEndOfTime` and user defined end time,
+// return the earlier one
+func computeActualEndTime(userInputEndTime string, endOfTime time.Time) time.Time {
+	var userDefinedEndOfTime time.Time
+	var err error
```

Comment:
consider removing both declarations and change L48 to userdefinedendoftime,err **:=** parseUnixtimestring().

Reply:
Thanks for the suggestion and knowledge sharing @kritikapradhan Updated as suggested